src/node_interface.rs

```diff
@@ -116,7 +116,8 @@ pub trait Interface : Sync + Send {
                         data: Vec<u8>) -> Result<MessageAction, InterfaceError>;
 }
 
-pub trait CreatePersonas<F : Interface> : Sync + Send  {
-    fn create_personas(&mut self) -> F;
```

Comment:
please start from master and don't pull in the commits from "simplifying routing_node creation"